src/unix/loop.c

```diff
@@ -68,6 +69,13 @@ int uv_loop_init(uv_loop_t* loop) {
   if (err)
     return err;
 
+  mloop = (uv__metrics_loop_t*) uv__calloc(1, sizeof(uv__metrics_loop_t));
```

Comment:
This call might fail. Take it into account.

Reply:
You are correct. I've moved this to near the top of the function and have it return `UV_ENOMEM` if it fails.